velox/exec/HashProbe.h

```diff
@@ -430,12 +430,20 @@ class HashProbe : public Operator {
   // Row number in 'input_' for each output row.
   BufferPtr outputRowMapping_;
 
+  // For left join with filter, we could overwrite the row which we have not
+  // checked if there is a carryover.  Use a temporary buffer in this case.
+  BufferPtr tempOutputRowMapping_;
+
   // maps from column index in 'table_' to channel in 'output_'.
   std::vector<IdentityProjection> tableOutputProjections_;
 
   // Rows of table found by join probe, later filtered by 'filter_'.
   std::vector<char*> outputTableRows_;
 
+  // For left join with filter, we could overwrite the row which we have not
+  // checked if there is a carryover.  Use a temporary buffer in this case.
+  BufferPtr tempOutputTableRows_;
```

Comment:
Any particular reason this type doesn't match 'outputTableRows_'?

Reply:
`std::vector<char*>` is untracked in memory pool so I just use a buffer here for it to be tracked.

Reply:
> std::vector<char*> is untracked in memory pool 

@Yuhta Should we "fix" that in a follow-up PR?

Reply:
Yeah we can do that